src/lib/core/common-behaviors/color.ts

```diff
@@ -0,0 +1,47 @@
+import {Constructor} from './constructor';
+import {ElementRef, Renderer2} from '@angular/core';
+
+/** List of possible color values that can be set. */
+const AVAILABLE_COLOR_VALUES = ['primary', 'accent', 'warn'];
+
+/** @docs-private */
+export interface IsColorable {
+  color: string;
+}
+
+/** @docs-private */
+export interface ColorableBase {
+  _renderer: Renderer2;
+  _elementRef: ElementRef;
+}
+
+/** Mixin to augment a directive with a `color` property. */
+export function mixinColor<T extends Constructor<ColorableBase>>(base: T, allowNoColor = false)
+    : Constructor<IsColorable> & T {
+  return class extends base {
+    private _color: string = null;
+
+    constructor(...args: any[]) { super(...args); }
+
+    get color(): string { return this._color; }
+    set color(value: string) {
+      if (AVAILABLE_COLOR_VALUES.indexOf(value) !== -1 || (allowNoColor && !value)) {
```

Comment:
Rather than having a run-time check, use string literal values for the type so that it's enforced at compile time:
```ts
protected _color: 'primary' | 'accent' | 'warn' | undefined = undefined;
```